### ISSUE TITLE
Cannot play any video

### ISSUE THREAD

**Issue by kuba-orlik** [anxdpanic/plugin.video.youtube#117]
### Context

* Add-on Version: 6.8.10
* Kodi Version: 18.7
* Kodi GUI Language: English
* Operating System: Raspbian
* Operating System Language: English

------

### Expected Behavior

Opening a video works

------

### Current Behavior

Opening a video shows a spinner for a while, and then an error shows up telling me to check logs for more details.

I also cannot open the plugin itself (the main view), it crashes immediately. Settings window works, though

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Share a video with "cast to kodi"
2. See the error message
3. The video doesn't play

### Log

Please include a complete [debug log](https://kodi.wiki/view/Log_file).

```
2021-05-06 22:06:50.423 T:1675591904   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: No JSON object could be decoded
                                            Traceback (most recent call last):
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/default.py", line 12, in <module>
                                                from youtube_plugin import youtube
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/__init__.py", line 11, in <module>
                                                from .provider import Provider
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/provider.py", line 22, in <module>
                                                from ..youtube.client import YouTube
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__init__.py", line 11, in <module>
                                                from .youtube import YouTube
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py", line 22, in <module>
                                                from .login_client import LoginClient
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/login_client.py", line 17, in <module>
                                                from .__config__ import api, youtube_tv, developer_keys, keys_changed
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__config__.py", line 222, in <module>
                                                _api_check = APICheck(__context, __settings)
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__config__.py", line 30, in __init__
                                                self._am_jstore = LoginTokenStore()
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/json_store/login_tokens.py", line 17, in __init__
                                                JSONStore.__init__(self, 'access_manager.json')
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/json_store/json_store.py", line 39, in __init__
                                                self.load()
                                              File "/home/pi/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/json_store/json_store.py", line 59, in load
                                                data = json.load(jsonfile)
                                              File "/usr/lib/python2.7/json/__init__.py", line 291, in load
                                                **kw)
                                              File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
                                                return _default_decoder.decode(s)
                                              File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
                                                obj, end = self.raw_decode(s, idx=_w(s, 0).end())
                                              File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
                                                raise ValueError("No JSON object could be decoded")
                                            ValueError: No JSON object could be decoded
                                            -->End of Python script error report<--
2021-05-06 22:06:50.476 T:1683984608   ERROR: GetDirectory - Error getting plugin://plugin.video.youtube/
2021-05-06 22:06:50.479 T:1899955120   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.youtube/) failed
```

------

### Additional Information

Please provide any additional information that may be helpful.

------

**Comment by RManPT** [comment]
One more, same issue. Both in Libreelec 9.2.x and 9.95.x

UPDATE:

Solved it.

Noticed that the problem has to do with some json that is not being parsed and found out that "/.kodi/userdata/addon_data/plugin.video.youtube/access_manager.json" was empty. Deleted it. Reran the addon and error was gone, a new file was created. Was unable to login though, so went to "Setting/Maintenance" tab and "Delete settings.xml", next went to the "API" tab and "Enable API configuration page", reran the addon and was asked to go to google.com/device to re-enable authorization for the addon to use my API keys (that were still there). Boom, after some settings tweaking it is as good as new!

**Comment by RManPT** [comment]
Well, solved the issue. I noticed the error was about parsing a json file that was not available so I messed around and found that "access_manager.json" was empty (.kodi/userdata/addon_data/plugin.video.youtube), it somehow got corrupted. I deleted both access_manager.json and api_keys.json, reran the addon and the error was gone. I was not able to login to my account though, so I went to addon settings and in "Maintenance" deleted setting.xml and activated the "Enable API configuration page" at the API tab and followed the instruction to re-allow the app to use my API keys (that were never really removed). Good as new!

Hope this helps.
